mindsdb_native/libs/phases/model_analyzer/model_analyzer.py

```diff
@@ -152,3 +157,70 @@ def predict_wrapper(*args, **kwargs):
+
+            fit_params = {'target': target,
+                          'all_columns': self.transaction.lmd['columns'],
+                          'columns_to_ignore': self.transaction.lmd['columns_to_ignore'] +
+                                               [col for col in output_columns if col != target]}
+
+            if is_classification:
+                if data_subtype != DATA_SUBTYPES.TAGS   :
+                    all_targets = [elt[1][target].values for elt in inspect.getmembers(self.transaction.input_data)
+                                   if elt[0] in {'test_df', 'train_df', 'validation_df'}]
+                    all_classes = np.unique(np.concatenate([np.unique(arr) for arr in all_targets]))
+
+                    enc = OneHotEncoder(sparse=False)
+                    enc.fit(all_classes.reshape(-1, 1))
+                    fit_params['one_hot_enc'] = enc
+                    self.transaction.hmd['label_encoders'][target] = enc
+                else:
+                    fit_params['one_hot_enc'] = None
+                    self.transaction.hmd['label_encoders'][target] = None
+
+                adapter = ConformalClassifierAdapter
+                nc_function = MarginErrFunc()  # better than IPS as we'd need the complete distribution over all classes
+                nc_class = ClassifierNc
+                icp_class = IcpClassifier
+
+            else:
+                self.transaction.lmd['stats_v2']['train_std_dev'][target] = self.transaction.input_data.train_df[target].std()
+                adapter = ConformalRegressorAdapter
+                nc_function = AbsErrorErrFunc()
+                nc_class = RegressorNc
+                icp_class = IcpRegressor
+
+            model = adapter(self.transaction.model_backend.predictor, fit_params=fit_params)
+            nc = nc_class(model, nc_function)
+            if is_classification:
+                self.transaction.hmd['icp'][target] = icp_class(nc, smoothing=False)
+            else:
+                self.transaction.hmd['icp'][target] = icp_class(nc)
+
+            if data_subtype != DATA_SUBTYPES.TAGS:
+                X = deepcopy(self.transaction.input_data.train_df)
+                y = X.pop(target)
+                [X.pop(col) for col in output_columns if col != target]
+                self.transaction.hmd['icp'][target].fit(X.values, y.values)
+
+                # calibrate conformal estimator on test set
+                X = deepcopy(self.transaction.input_data.validation_df)
+                y = X.pop(target).values
+                [X.pop(col) for col in output_columns if col != target]
+
+                if is_classification:
+                    if isinstance(enc.categories_[0][0], str):
+                        cats = enc.categories_[0].tolist()
+                        y = np.array([cats.index(i) for i in y])
+                    y = y.astype(int)
+
+                self.transaction.hmd['icp'][target].calibrate(X.values, y)
```

Comment:
Is this going to be able to work when the input data contains types such as `text` or `image` or `array` ?